tools/istio-iptables/pkg/builder/iptables_builder_impl.go

```diff
@@ -170,6 +177,120 @@ func (rb *IptablesRuleBuilder) buildRules(rules []*Rule) [][]string {
 	return output
 }
 
+func reverseRules(rules []*Rule) []*Rule {
```

Comment:
I know we have 1-2 bigger scoped tests that you added, and the existing golden-based checks, but I would like (smaller, simpler, direct) unit tests around these specific mutative funcs, directly testing the logic branches present in each.


Especially since all these helper funcs are 

- Critical
- Have no hard OS deps/are all string manip

it would be much better to describe the behavior of these functions in tighter unit tests with no deps/mocks/extra layers.

Reply:
Sure, makes sense. I will work on adding those :+1: